app/services/notifications/new_comment/send.rb

```diff
@@ -17,27 +17,40 @@ def call
         return if comment.score.negative?
 
         user_ids = Set.new(comment_user_ids + subscribed_user_ids + top_level_user_ids + author_subscriber_user_ids)
+        user_ids.delete(comment.user_id)
 
         json_data = {
           user: user_data(comment.user),
           comment: comment_data(comment)
         }
 
-        targets = []
-        user_ids.delete(comment.user_id).each do |user_id|
+        user_ids.each do |user_id|
           Notification.create(
             user_id: user_id,
             notifiable_id: comment.id,
             notifiable_type: comment.class.name,
             action: nil,
             json_data: json_data,
           )
-
-          targets << "user-notifications-#{user_id}" if User.find_by(id: user_id)&.mobile_comment_notifications
         end
 
-        # Sends the push notification to Pusher Beams channels. Batch is in place to respect Pusher 100 channel limit.
-        targets.each_slice(100) { |batch| send_push_notifications(batch) }
+        targets = User.where(id: user_ids, mobile_comment_notifications: true).order(:id).ids
```

Comment:
tiny non blocking detail: is the ordering important?

Reply:
Functionality-wise the ordering is not important, but I believe that without it the query does not guarantee a deterministic order of the ids. When I pushed a few of the most recent commits I ran into failures of [this test](https://github.com/forem/forem/pull/13036#discussion_r596945341) because the ids weren't always returned in the same order. 

So, it avoids adding a flaky spec that other contributors could run into in the future. Also, that test will be removed when we finish working on RFC 68 (when we remove the Beams dependency and logic altogether).

Since it's all within a single DB query I don't think it should be too costly performance-wise (also considering this is reducing the N queries performed before on individual users vs one query now).

Reply:
> Functionality-wise the ordering is not important, but I believe that without it the query does not guarantee a deterministic order of the ids. When I pushed a few of the most recent commits I ran into failures of this test because the ids weren't always returned in the same order.

ah ok, you can use `match_array()` in Rspec for that, it doesn't take order into consideration

> Since it's all within a single DB query I don't think it should be too costly performance-wise (also considering this is reducing the N queries performed before on individual users vs one query now).

yeah, you're right. One other thing I just notied: `mobile_comment_notifications` doesn't have an index, we might want to add it...

Reply:
> ah ok, you can use match_array() in Rspec for that, it doesn't take order into consideration

Awesome, thanks for the suggestion! I'll try that and get rid of the order in that case (no actual need to keep that order clause)

> yeah, you're right. One other thing I just notied: mobile_comment_notifications doesn't have an index, we might want to add it...

#12768 is currently refactoring the user settings, so I'll wait for that refactor to be merged first. I [added this](https://github.com/orgs/forem/projects/28#card-57298336) to our Project board so we don't forget 😃